runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -243,9 +363,277 @@ TR::ARM64PrivateLinkage::setParameterLinkageRegisterIndex(TR::ResolvedMethodSymb
    }
 
 
+int32_t
+TR::ARM64PrivateLinkage::calculatePreservedRegisterSaveSize(
+      uint32_t &registerSaveDescription,
+      uint32_t &numGPRsSaved)
+   {
+   TR::Machine *machine = cg()->machine();
+
+   TR::RealRegister::RegNum firstPreservedGPR = TR::RealRegister::x21;
+   TR::RealRegister::RegNum lastPreservedGPR = TR::RealRegister::x28;
+
+   // Create a bit vector of preserved registers that have been modified
+   // in this method.
+   //
+   for (int32_t i = firstPreservedGPR; i <= lastPreservedGPR; i++)
+      {
+      if (machine->getRealRegister((TR::RealRegister::RegNum)i)->getHasBeenAssignedInMethod())
+         {
+         registerSaveDescription |= 1 << (i-1);
+         numGPRsSaved++;
+         }
+      }
+
+   return numGPRsSaved*8;
+   }
+
+
 void TR::ARM64PrivateLinkage::createPrologue(TR::Instruction *cursor)
    {
-   TR_UNIMPLEMENTED();
+
+   // Prologues are emitted post-RA so it is fine to use real registers directly
+   // in instructions
+   //
+   TR::ARM64LinkageProperties& properties = getProperties();
+   TR::Machine *machine = cg()->machine();
+   TR::RealRegister *vmThread = machine->getRealRegister(properties.getMethodMetaDataRegister());   // x19
+   TR::RealRegister *javaSP = machine->getRealRegister(properties.getStackPointerRegister());       // x20
+
+   // 'interpretedMethodEntryPoint' and 'jittedMethodEntryPoint` currently only
+   // exist to document the meaning of certain instruction cursor values.  They
+   // are expected to be easily optimized away by a build compiler.
+   //
+   TR::Instruction *interpretedMethodEntryPoint = cursor;
+
+   // --------------------------------------------------------------------------
+   // Create the entry point when transitioning from an interpreted method.
+   // Parameters are passed on the stack, so load them into the appropriate
+   // linkage registers expected by the JITed method entry point.
+   //
+   cursor = loadStackParametersToLinkageRegisters(cursor);
+
+   TR::Instruction *jittedMethodEntryPoint = cursor;
```

Comment:
How do you encode the distance between `jittedMethodEntryPoint` and `interpretedMethodEntryPoint` into the linkage info word?
Other architectures do that in `CodeGenerator::doBinaryEncoding()`.